feature-libs/checkout/base/core/facade/checkout-payment.service.spec.ts

```diff
@@ -0,0 +1,338 @@
+import { AbstractType, Type } from '@angular/core';
+import { fakeAsync, inject, TestBed, tick } from '@angular/core/testing';
+import { MockStore, provideMockStore } from '@ngrx/store/testing';
+import {
+  CheckoutQueryFacade,
+  CheckoutState,
+  PaymentDetailsCreatedEvent,
+  PaymentDetailsSetEvent,
+} from '@spartacus/checkout/base/root';
+import {
+  ActiveCartService,
+  CardType,
+  Cart,
+  EventService,
+  HttpErrorModel,
+  OCC_USER_ID_CURRENT,
+  PaymentDetails,
+  QueryState,
+  UserActions,
+  UserIdService,
+} from '@spartacus/core';
+import { Observable, of, throwError } from 'rxjs';
+import { take } from 'rxjs/operators';
+import { CheckoutPaymentConnector } from '../connectors/payment/checkout-payment.connector';
+import { CheckoutPaymentService } from './checkout-payment.service';
+
+const mockUserId = OCC_USER_ID_CURRENT;
+const mockCartId = 'cartID';
+const mockJaloError: Partial<HttpErrorModel> = {
+  details: [{ type: 'JaloObjectNoLongerValidError' }],
+};
+const mockCardTypes: CardType[] = [
+  {
+    code: 'VISA',
+    name: 'Visa',
+  },
+  {
+    code: 'MASTER',
```

Comment:
you have to change this to MASTERCARD :P, our thing is flagging the language

Reply:
🙈